### ISSUE TITLE
AppSettings: Passcode lock

### ISSUE THREAD

**Issue by jesmrec** [owncloud/ios-app#25]
_As an user, i want to lock my app with a passcode, so that i can ensure that no one will reach my files_

AC:

- [ ] Passcode will have 4 digits. 
- [ ] It has to be input twice. If second time does not match the first one, error displayed
- [ ] Possibility to cancel. 
- [ ] To remove the passcode, is a must to enter the code
- [ ] Passcode always asked to enter the app
- [ ] Passcode always asked from Files App / 3rd party
- [ ] Set a time after asking the passcode, in which the passcode will not be asked
- [ ] QA

**Comment by javiergonzper** [comment]
I am working on the design of the passcode using the current Theme styles

<img width="246" alt="screen shot 2018-05-03 at 14 57 44" src="https://user-images.githubusercontent.com/1815050/39577814-99f833b0-4ee2-11e8-80ae-20cbdeef06fa.png">

<img width="246" alt="screen shot 2018-05-03 at 14 57 57" src="https://user-images.githubusercontent.com/1815050/39577815-9a1367c0-4ee2-11e8-97c3-832118d0e28d.png">

I hope you like it 😄

**Comment by theScrabi** [comment]
@javiergonzper I honestly don't like that. I don't know much about the IPhone design guidelines, but I'm pretty sure this input field does not align them.

I set up an alternative design. Please tell me what you think of it:

<img width="246" alt="antyeyecancer_white" src="https://user-images.githubusercontent.com/1891273/39697151-419442ac-51f0-11e8-9179-7145a8df669f.png"> <img width="246" alt="antyeyecancer_black" src="https://user-images.githubusercontent.com/1891273/39697154-43d3e072-51f0-11e8-8296-d6f3b5ec4bb0.png">

**Comment by jesmrec** [comment]
@michaelstingl @felix-schwarz @pablocarmu 

- round buttons vs square buttons
- separate digits vs all digits together

one question: why are used custom buttons instead of the built-in keyboard? 

i like round buttons, but also i would rather all digits together.

**Comment by michaelstingl** [comment]
Examples from other apps:
https://github.com/owncloud/ios-app/issues/3#issuecomment-385339469

**Comment by michaelstingl** [comment]
This is the native (iOS 11) look and feel:

| - | 4-digit numeric | 6-digit numeric | Custom numeric |
|-|-|-|-|
| Lock screen | ![2018-05-07 17 11 59](https://user-images.githubusercontent.com/214010/39710993-b43d76ce-521e-11e8-8188-b925d5b1d41c.png) | ![2018-05-07 17 14 59](https://user-images.githubusercontent.com/214010/39711008-bd9dd3a8-521e-11e8-9ce6-91f318253f82.png) | ![2018-05-07 17 18 57](https://user-images.githubusercontent.com/214010/39711016-c3fc0e90-521e-11e8-9610-0fbeb3655576.png) |
| Settings screen | ![2018-05-07 17 12 25](https://user-images.githubusercontent.com/214010/39711032-ce4c21d2-521e-11e8-91b5-9d04b5362706.png) | ![2018-05-07 17 13 05](https://user-images.githubusercontent.com/214010/39711046-d477e87a-521e-11e8-811f-7081981c6217.png) | ![2018-05-07 17 13 37](https://user-images.githubusercontent.com/214010/39711055-db5d5c10-521e-11e8-86a3-4b6f6361286b.png) |

**Comment by javiergonzper** [comment]
> Android app has separate input fields for each digit. I think it better that way.

I did all digits together because it could be parametrizable.
`let numberDigitsPasscode = 4`
Right now we use a 4 digits passcode but maybe in the future we will need a 6 digits (or more) and we just have to change that value.

About the round buttons. My idea was use round buttons but when I apply the OC button styles with the corners I liked it and because of that I let them as you see.

I used the numbers because is the way that the user unlock his iPhone every time. The design with the keyboard of the "Settings" it is aligned with all the "Settings" inteface.

The design that I made it is just now developed:

<img width="246" alt="screen shot 2018-05-08 at 08 19 29" src="https://user-images.githubusercontent.com/1815050/39741043-7bf806b0-5298-11e8-87b3-22e122f4864b.png">
<img width="246" alt="screen shot 2018-05-08 at 08 19 45" src="https://user-images.githubusercontent.com/1815050/39741044-7c21e5ac-5298-11e8-8d66-097d88e6a55d.png">
